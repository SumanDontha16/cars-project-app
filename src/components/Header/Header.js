import React from "react";

const Header = () => {
	return (
		<>
			<nav
				className="navbar navbar-light py-4 mb-4"
				style={{ backgroundColor: "#24356d" }}
			>
				<div className="container">
					<h4
						className="display-6"
						style={{ color: "#ffffff" }}
						data-testid="header"
					>
						Cars Inventory
					</h4>
				</div>
			</nav>
		</>
	);
};

export default Header;
