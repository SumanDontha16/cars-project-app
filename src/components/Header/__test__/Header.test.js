import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "../Header";
import renderer from "react-test-renderer";

describe("Header", () => {
	it("Test to match snapshot of component", () => {
		const headerTree = renderer.create(<Header />).toJSON();
		expect(headerTree).toMatchSnapshot();
	});

	it("Header renders with correct title", () => {
		render(<Header />);
		const headerEl = screen.getByTestId("header");

		expect(headerEl.textContent).toBe("Cars Inventory");
	});
});
