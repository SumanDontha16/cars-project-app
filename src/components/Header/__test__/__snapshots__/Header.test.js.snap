// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test to match snapshot of component 1`] = `
<nav
  className="navbar navbar-light py-4 mb-4"
  style={
    Object {
      "backgroundColor": "#24356d",
    }
  }
>
  <div
    className="container"
  >
    <h4
      className="display-6"
      data-testid="header"
      style={
        Object {
          "color": "#ffffff",
        }
      }
    >
      Cars Inventory
    </h4>
  </div>
</nav>
`;
